### PR TITLE
master merge

### PR DESCRIPTION
**Pull request by ieow**

### CODE REVIEW

File: src/mpcCoreKit.ts [hunk 1]
```diff
@@ -427,13 +428,15 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
...
-
       const hashedFactorPub = getPubKeyPoint(hashedFactorKey);
       await this.deleteFactor(hashedFactorPub, hashedFactorKey);
       await this.deleteShareWithFactorKey(hashedFactorKey);
 
-      return backupFactorKey;
+      // only recovery factor = true
+      if (recoveryFactor) {
+        const backupFactorKey = await this.createFactor({ shareType: TssShareType.RECOVERY, ...enableMFAParams });
+        return backupFactorKey;
```
Comment (yashovardhan) [inline src/mpcCoreKit.ts:438]: If someone passes recovery factor to be `false` and clear browser storage, they are at a risk of losing their account right?

File: src/helper/browserStorage.ts [hunk 2]
```diff
@@ -68,6 +68,12 @@ export class BrowserStorage {
     store[key] = value;
     this.storage.setItem(this._storeKey, JSON.stringify(store));
   }
+
+  remove(key: string): void {
+    const store = JSON.parse(this.storage.getItem(this._storeKey) || "{}");
+    delete store[key];
+    this.storage.setItem(this._storeKey, JSON.stringify(store));
+  }
```
Comment (yashovardhan) [inline src/helper/browserStorage.ts:76]: Should we also add a function for deleted web browser factor additionally for ease of use?

**Review by yashovardhan — COMMENTED** [review]
Added a few comments, otherwise good to go.

File: src/mpcCoreKit.ts (hunk 1 above)
Comment (ieow) [reply src/mpcCoreKit.ts:438]: Yes, this is requested by safe to be not to opinionated

Comment (yashovardhan) [reply src/mpcCoreKit.ts:438]: ok

File: src/helper/browserStorage.ts (hunk 2 above)
Comment (ieow) [reply src/helper/browserStorage.ts:76]: I am thinking add a clear which clear all related browser data

Comment (yashovardhan) [reply src/helper/browserStorage.ts:76]: makes sense

Comment (ieow) [reply src/helper/browserStorage.ts:76]: There is resetStore func for clearing

Comment (yashovardhan) [reply src/helper/browserStorage.ts:76]: oh right, cool then good to go.

**Review by yashovardhan — APPROVED** [review]
LGTM

File: src/interfaces.ts [hunk 3]
```diff
@@ -155,6 +155,7 @@ export interface ICoreKit {
   /**
    * The function used to initailise the state of MPCCoreKit
    * Also is useful to resume an existing session.
+   * @param params - handleRedirectResult: boolean - flag for handle the redirect result - Default is true.
    */
   init(): Promise<void>;
```
Comment (matthiasgeihs) [inline src/interfaces.ts:160]: You are adding a parameter to the function doc without adding the parameter to the interface...

File: src/interfaces.ts [hunk 4]
```diff
@@ -173,6 +174,8 @@ export interface ICoreKit {
   /**
    * Enable MFA for the user. Deletes the Cloud Factor and generates a new Factor Key.
    * Recommended for Non Custodial Flow.
+   * @param enableMFAParams - Parameters for recovery factor for MFA.
+   * @param recoveryFactor - Default is true. If false, recovery factor will NOT be created.
    */
   enableMFA(enableMFAParams: EnableMFAParams): Promise<string>;
```
Comment (matthiasgeihs) [inline src/interfaces.ts:180]: You are adding a parameter to the function doc without adding the parameter to the interface...

File: src/mpcCoreKit.ts [hunk 5]
```diff
@@ -175,7 +175,7 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     return this.options.uxMode === UX_MODE.REDIRECT;
   }
 
-  public async init(): Promise<void> {
+  public async init(params: { handleRedirectResult: boolean } = { handleRedirectResult: true }): Promise<void> {
```
Comment (matthiasgeihs) [inline src/mpcCoreKit.ts:178]: should define type for `{ handleRedirectResult: boolean }` as this is part of the external interface. Or just use `handleRedirectResult` as single parameter...

**Pull request merged**